sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/StargateBridgeClient.java

```diff
@@ -60,17 +60,25 @@ default Response executeBatch(Batch batch) {
   /**
    * Gets the metadata describing the given keyspace.
    *
-   * @throws UnauthorizedKeyspaceException (wrapped in the returned future) if the client is not
-   *     allowed to access this keyspace. However, this check is shallow: if the caller surfaces
-   *     keyspace elements (tables, UDTs...) to the end user, it must authorize them individually
-   *     with {@link #authorizeSchemaReads(List)}.
+   * @param checkIfAuthorized whether to check if the current user is allowed to describe this
+   *     keyspace (this requires an additional background call to the bridge). This is an
+   *     optimization: if you only use the schema metadata to build a DML query that will be
+   *     executed immediately after, you can skip authorization since {@link
+   *     #executeQueryAsync(Query)} does it automatically. However, if there is no other query and
+   *     you build a client response directly from the metadata, then authorization should be done
+   *     by this method.
+   * @throws UnauthorizedKeyspaceException (wrapped in the returned future) if {@code
+   *     checkIfAuthorized} is set, and the client is not allowed to access this keyspace. However,
+   *     this check is shallow: if the caller surfaces keyspace elements (tables, UDTs...) to the
+   *     end user, it must authorize them individually with {@link #authorizeSchemaReads(List)}.
    */
-  CompletionStage<Optional<CqlKeyspaceDescribe>> getKeyspaceAsync(String keyspaceName);
+  CompletionStage<Optional<CqlKeyspaceDescribe>> getKeyspaceAsync(
+      String keyspaceName, boolean checkIfAuthorized);
```

Comment:
I ended up adding a parameter for authorization. Having a default was not intuitive either way, I like that this forces the caller to think about it.